log: Remove brittle and confusing LogPrintLevel

maflcko opened the pull request:
`LogPrintLevel` has many issues:

* It encourages to log several levels in one source location. This is problematic, because all levels (even warnings and errors) will be rate limited equally for the same location.
* Its warning and error logs are specially formatted compared to all other warning and error logs in the codebase, making them harder to spot (both in the debug log and in the code).
* It is verbose to type and read.
* It is confusing, because the majority of code uses the `Log$LEVEL(...)` macros. Having less ways to achieve the same makes the code more consistent and easier to review.

Fix all issues by removing it

stickies-v committed:
libevent: separate log statements per level

Avoids ratelimiting unconditional log statements when debug logging
is enabled. Introduces slight behaviour change by removing
the category from unconditional logs, making them more uniform
with the other unconditional logs in the codebase.

Also, in a slight behavior change, prefix the info-level (and higher)
messages with "libevent:".

stickies-v committed:
ipc: separate log statements per level

Avoids ratelimiting unconditional log statements when debug logging
is enabled. Introduces slight behaviour change by removing
the category from unconditional logs, making them more uniform
with the other unconditional logs in the codebase.

Also, in a slight behavior change, prefix the info-level (and higher)
messages with "ipc:".

Commit:
log: Remove brittle and confusing LogPrintLevel

stickies-v reviewed (approved):
ACK fa35682637e7848106189e314a963d8f6c45b2bc

w0xlt reviewed (commented):
ACK https://github.com/bitcoin/bitcoin/pull/34051/commits/fa35682637e7848106189e314a963d8f6c45b2bc

Ataraxia009 reviewed (commented):
Concept Ack

maflcko commented on src/test/logging_tests.cpp:
```diff
@@ -207,22 +189,20 @@ BOOST_FIXTURE_TEST_CASE(logging_SeverityLevels, LogSetup)
-    LogPrintLevel(BCLog::VALIDATION, BCLog::Level::Warning, "foo3: %s\n", "bar3");
-    LogPrintLevel(BCLog::RPC, BCLog::Level::Error, "foo4: %s\n", "bar4");
+    LogInfo("info_%s", 1);
+    LogTrace(BCLog::HTTP, "trace_%s. This log level is lower than the global one.", 2);
```
I've added a new `LogDebug(BCLog::HTTP,` in the line below, to clarify this in the test itself. If that is not sufficient, I am happy to push any diff or commit here, if someone writes one. Also, I am happy to review a separate pull, or a follow-up.

maflcko replied:
Thanks for providing a diff. I've pushed a commit with that:

```diff
$ git diff fa35682637 fa33a9f5be -U0
diff --git a/src/test/logging_tests.cpp b/src/test/logging_tests.cpp
index 3d95c97426..2a34aefae1 100644
--- a/src/test/logging_tests.cpp
+++ b/src/test/logging_tests.cpp
@@ -187,0 +188 @@ BOOST_FIXTURE_TEST_CASE(logging_SeverityLevels, LogSetup)
+    LogInstance().SetLogLevel(BCLog::Level::Debug);

Commit:
test: Clarify logging_SeverityLevels test

The test was a bit confusing, because it just referred to the "global
log level" without explicitly specifying what it is. The level is set
though the LogSetup constructor. However, it is easier to follow unit
tests, if they are self-contained. So just set the level to Debug
explicitly here.

Also, add a new debug_3 log, to further document the intended behavior
of the unit test.

Also, replace the LogPrintLevel with the shorter and exact replacements
LogTrace and LogDebug.

rkrux reviewed (approved):
ACK fa33a9f5be1beabfee6c217d133cc51dea3b9a63

This is helpful. Having gone through the logging code recently, I found myself confused a bit upon seeing different ways to log. Removal of `LogPrintLevel` helps in decreasing the confusion by just using `LOG<$LEVEL>()` macros.

Ataraxia009 commented on src/logging.h:
```diff
@@ -372,22 +373,20 @@ inline void LogPrintFormatInternal(std::source_location&& source_loc, BCLog::Log
+// the category is allowed. No rate limiting is applied, because users specifying -debug are assumed to be
 // developers or power users who are aware that -debug may cause excessive disk usage due to logging.
-#define LogPrintLevel(category, level, ...)                           \
+#define detail_LogIfCategoryAndLevelEnabled(category, level, ...)     \
```
I still don't see the point of leaving the `Assume` in there since it seems to make no difference other than constricting somebody that would want to create a new macro for `LogInfo` w category. But i wont hold the PR back because of it.

pablomartin4btc reviewed (commented):
ACK fa33a9f5be1beabfee6c217d133cc51dea3b9a63

I agree with the benefits mentioned in the PR description.

pablomartin4btc reviewed (approved):
re-ACK fa8a5d215c5a81a7282fd5dd1098f9d3fa40e5db

vasild commented on src/mapport.cpp:
```diff
@@ -47,7 +47,7 @@ static void ProcessPCP()
     auto handle_mapping = [&](std::variant<MappingResult, MappingError> &res) -> void {
         if (MappingResult* mapping = std::get_if<MappingResult>(&res)) {
-            LogPrintLevel(BCLog::NET, BCLog::Level::Info, "portmap: Added mapping %s\n", mapping->ToString());
+            LogInfo("portmap: Added mapping %s", mapping->ToString());
```
Here and elsewhere these messages had a category before and now they are category-less. When a message has a category its log is prefixed by e.g. `[net:info]` which makes it easier to filter/grep all "net" messages and to know which module a message comes from.

Would be nice to have the ability to log info/warning/error messages with category.
